iroh-relay/src/client.rs

```diff
@@ -143,45 +141,21 @@ impl ClientBuilder {
 
     /// Establishes a new connection to the relay server.
     pub async fn connect(&self) -> Result<Client> {
-        let roots = rustls::RootCertStore {
-            roots: webpki_roots::TLS_SERVER_ROOTS.to_vec(),
-        };
-        let mut config = rustls::client::ClientConfig::builder_with_provider(Arc::new(
-            rustls::crypto::ring::default_provider(),
-        ))
-        .with_safe_default_protocol_versions()
-        .expect("protocols supported by ring")
-        .with_root_certificates(roots)
-        .with_no_client_auth();
-        #[cfg(any(test, feature = "test-utils"))]
-        if self.insecure_skip_cert_verify {
-            warn!("Insecure config: SSL certificates from relay servers not verified");
-            config
-                .dangerous()
-                .set_certificate_verifier(Arc::new(NoCertVerifier));
-        }
-        config.resumption = Resumption::default();
-        let tls_connector: tokio_rustls::TlsConnector = Arc::new(config).into();
```

Comment:
Instead of immediately constructing the `tls_connector` in `connect`, and then either using it or discarding it depending on whether we use `connect_ws` or `connect_relay`, I've moved it into `connect_relay`.